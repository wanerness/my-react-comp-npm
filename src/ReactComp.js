import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { trim } from './utils/util';

class ReactComp extends Component {
  render() {
    const { onClick, title } = this.props;
    return (
      <div>
        <button onClick={onClick}>{trim(title)}</button>
      </div>
    );
  }
}

ReactComp.propTypes = {
  title: PropTypes.string,
  onClick: PropTypes.func
};

export default ReactComp;
