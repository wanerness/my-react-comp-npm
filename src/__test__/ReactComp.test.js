import React from 'react';
import renderer from 'react-test-renderer';

import ReactComp from '../ReactComp';

describe('test ReactComp', () => {
  it('renders ReactCompe button', () => {
    const tree = renderer
      .create(
        <ReactComp
          title='fku'
          onClick={() => {
            console.log('click me');
          }}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
