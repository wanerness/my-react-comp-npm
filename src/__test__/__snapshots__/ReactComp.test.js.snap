// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ReactComp renders ReactCompe button 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    fku
  </button>
</div>
`;
